src/main/java/com/liferay/faces/util/i18n/internal/I18nImpl.java

```diff
@@ -53,10 +55,27 @@ public I18nImpl() {
 
 		// Store the resource bundle cache in the application map (as a Servlet Context attribute).
 		if (startupFacesContext != null) {
+
 			ExternalContext externalContext = startupFacesContext.getExternalContext();
 			Map<String, Object> applicationMap = externalContext.getApplicationMap();
-			Map<Locale, ResourceBundle> cache = new ConcurrentHashMap<Locale, ResourceBundle>();
-			applicationMap.put(I18nImpl.class.getName(), cache);
+			Cache<Locale, ResourceBundle> facesResourceBundleCache;
+
+			String maxCacheCapacityString = externalContext.getInitParameter(I18n.class.getName() +
+					".maxCacheCapacity");
+
+			if (maxCacheCapacityString != null) {
+
+				CacheFactory cacheFactory = (CacheFactory) FactoryExtensionFinder.getFactory(externalContext,
+						CacheFactory.class);
+				int initialCacheCapacity = cacheFactory.getDefaultInitialCapacity();
```

Comment:
@stiemannkj1 I realize that there might be a static vs. non-static method signature clash, but if this is a general pattern of being able to configure a maximum cache capacity then in order to eliminate boilerplate we should try and create a `static getDefaultInitialCapacity()` equivalent.